src/nodes/chunk_append/exec.c

```diff
@@ -256,6 +263,43 @@ do_startup_exclusion(ChunkAppendState *state)
 			}
 			restrictinfos = constify_restrictinfos(&root, restrictinfos);
 
+#if PG14_GE
+			// this is where we run the OSM chunk exclusion code
```

Comment:
I think this code should be moved inside can_exclude_chunk
can_exclude_chunk will do the usual predicate_refuted_by for the chunk AND if it is an OSM chunk, it will execute this additional logic.

Then all callers of can_exclude_chunk will do the right thing.
WDYT @zilder @konskov